cpp/oneapi/dal/backend/interop/table_conversion.hpp

```diff
@@ -57,6 +62,22 @@ inline auto convert_to_daal_homogen_table(array<T>& data,
                                                                  row_count);
 }
 
+#ifdef ONEAPI_DAL_DATA_PARALLEL
+template <typename T>
+inline auto convert_to_daal_sycl_homogen_table(sycl::queue& queue,
+                                               array<T>& data,
+                                               std::int64_t row_count,
+                                               std::int64_t column_count) {
+    data.need_mutable_data(queue);
+    const auto daal_data =
+        daal::services::SharedPtr<T>(data.get_mutable_data(), daal_array_owner<T>{ data });
+    return daal::data_management::SyclHomogenNumericTable<T>::create(daal_data,
+                                                                     column_count,
+                                                                     row_count,
+                                                                     cl::sycl::usm::alloc::shared);
```

Comment:
It's OK for now because we support only shared USM, but it is easy to forget to change this when we make support for device and host USM. Please, put TODO comment here or use `sycl::get_pointer_type()`: https://github.com/intel/llvm/blob/sycl/sycl/include/CL/sycl/usm.hpp#L163

Reply:
I agree. thanks